src/propagate-job.ts

```diff
@@ -90,19 +93,40 @@ export async function run(): Promise<void> {
         type: 2,
       };
 
-      // We populate the transactions we will use in our bundles
+      const _connectors: string[] = [];
+      const _encodedData: string[] = [];
+      const _fees: string[] = [];
+      let _totalFee = constants.Zero;
+
+      for (const domain of domains) {
+        const connector = rootManagerMeta.connectors[domains.indexOf(domain)];
+        _connectors.push(connector);
+
+        if (Object.keys(getParamsForDomainFn).includes(domain)) {
+          const getParamsForDomain = getParamsForDomainFn[domain];
+          const propagateParam = await getParamsForDomain(setup);
+          _encodedData.push(propagateParam._encodedData);
+          _fees.push(propagateParam._fee);
+          _totalFee = _totalFee.add(BigNumber.from(propagateParam._fee));
+        } else {
+          _encodedData.push('0x');
+          _fees.push('0');
+        }
+      }
+
+      // encode data for relayer proxy hub
+      const fee = BigNumber.from(0);
+
+      // We populate the transactions we will use
       const txs: TransactionRequest[] = await populateTransactions({
         chainId: CHAIN_ID,
-        contract: job as Contract,
-        functionArgs: [[]],
-        functionName: 'updateDeposits',
+        contract: mainnetSdk.relayerProxyHub,
+        functionArgs: [[_connectors, _fees, _encodedData, fee]],
+        functionName: 'propagate',
         options,
       });
 
-      // We calculate the first block that the first bundle in our batch will target.
-      // Example, if future blocks is 2, and we are in block 100, it will send a bundle to blocks 102, 103, 104 (assuming a burst size of 3)
-      // and 102 would be the firstBlockOfBatch
-      const firstBlockOfBatch = block.number + FUTURE_BLOCKS;
+      const firstBlockOfBatch = block.number;
```

Comment:
if applying my suggestion above, this should be: `firstBlockOfBatch = block.number + FUTURE_BLOCKS`